services:
  traefik:
    image: traefik:v3.1
    command: 
      - "--log.level=DEBUG"
      - "--api.insecure=true" 
      - "--providers.docker=true"
      - "--providers.docker.network=traefik_network"
      - "--providers.docker.exposedbydefault=false"
      - "--metrics.prometheus=true"
      - "--entryPoints.web.address=:80"
      - "--entryPoints.websecure.address=:443"
      - "--certificatesresolvers.myresolver.acme.httpchallenge=true"
      - "--certificatesresolvers.myresolver.acme.httpchallenge.entrypoint=web"
      - "--certificatesresolvers.myresolver.acme.email=mixmyxxe@gmail.com"
      - "--certificatesresolvers.myresolver.acme.storage=/letsencrypt/acme.json"
      - "--certificatesresolvers.myresolver.acme.caserver=https://acme-v02.api.letsencrypt.org/directory"
      - "443:443"
      - "8080:8080"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - ./letsencrypt:/letsencrypt
    networks:
      - traefik_network

  api:
    image: ghcr.io/drifterz13/nest-vultr-deploy:latest    
    environment:
      - PORT=8000
      - DATABASE_URL=file:/app/db/prod.db
    expose:
      - "8000"
    volumes:
      - db_data:/app/db
      - api_logs:/app/log
    networks:
      - traefik_network
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.api.rule=Host(`api.drifterz13.com`)"
      - "traefik.http.services.api.loadbalancer.server.port=8000"
      - "traefik.http.routers.api.entrypoints=websecure"
      - "traefik.http.routers.api.tls=true"
      - "traefik.http.routers.api.tls.certresolver=myresolver"
      - "traefik.http.services.api.loadbalancer.healthcheck.path=/health"
      - "traefik.http.services.api.loadbalancer.healthcheck.interval=10s"

volumes:
  db_data:
  api_logs:
    external: true
networks:
  traefik_network:
    external: true
